llpc/test/shaderdb/gfx10/CheckFMFOptions_NoContract.pipe

```diff
@@ -0,0 +1,220 @@
+// This test checks that setting disableFastMathFlags to 32 (1<<5 == AllowContract)
+// does actually stop contract (no fma instructions are formed).
+// See CheckFMFOptions_default.pipe for the default case.
+
+; BEGIN_SHADERTEST
+; RUN: amdllpc %gfxip %s -v | FileCheck -check-prefix=SHADERTEST %s
+; SHADERTEST-LABEL: _amdgpu_ps_main:
+; SHADERTEST: s_waitcnt
+; SHADERTEST:	v_mul_f32_e64 v[[ACC1:[0-9]*]], s{{[0-9]*}}, s{{[0-9]*}}
+; SHADERTEST-DAG:	v_mul_f32_e64 v[[ACC2:[0-9]*]], s{{[0-9]*}}, s{{[0-9]*}}
+; SHADERTEST-DAG:	v_add_f32_e32 v[[ACC1]], s{{[0-9]*}}, v[[ACC1]]
+; SHADERTEST-DAG:	v_add_f32_e32 v[[ACC2]], s{{[0-9]*}}, v[[ACC2]]
+; END_SHADERTEST
+
+[Version]
+version = 53
+
+[FsSpirv]
```

Comment:
Could you make the spir-v smaller, or just use the GLSL?  Spir-v is hard to read.  I can see what it is suppose to do because it contains the embedded GLSL.  However, the GLSL and spir-v could diverge if people have to update the test in the future.  Could a simple test where the body of the function contains 3 OpLoad instructions, 1 OpFMul, 1 OpFAdd, and one OpStore be good enough?  Instead, we have to follow lot of instructions to be able to see that there is a multiple and an add.

FYI: there is a tool called spirv-reduce that can help with this: https://github.com/KhronosGroup/SPIRV-Tools/#reducer.

Reply:
Yes, I'm just rewriting it now. I'm going to use the glsl directly, so no spirv required and reducing it to a single test file (the default option achieved by stripping the option out with grep -v like some of the other tests already do).
Also trying to add something that can verify that the cache hash is definitely different - but not sure how useful that actually is.

Reply:
Done